ARMmbed/mbedtls
# pkcs5: add support for additional hmacSHA algorithms

Currently only SHA1 is supported as hmac algorithm for PBKDF2
in PKCS5. This means that keys encrypted and authenticated using
another algorithm of the SHA family cannot be decrypted.

This deficiency has become particularly incumbent now that
PKIs created with openssl-1.1 are encrypting keys using
hmacSHA256.

Enable support for all the digest algorithms of the SHA
family for PKCS5.

Signed-off-by: Antonio Quartulli <antonio@openvpn.net>

## Status
**READY**

## Requires Backporting
NO  

## Migrations
NO

## Todos

## Steps to test or reproduce
Try to import a pk created and encrypted using openssl-1.1 and observe failure. With this PR the import succeeds.
